includes/DataObjects/User.php

```diff
@@ -644,7 +644,7 @@ public function isAdmin()
     
 	public function isCheckuser()
 	{
-		return $this->checkuser == 1 || $this->oauthCanCheckUser();
+		return $this->checkuser == 1 || ($this->isOAuthLinked() && $this->oauthCanCheckUser());
```

Comment:
Looking at the definition of oauthCanCheckUser(), I don't see that this actually changes anything.  What recursion does this pull request fix?